Authors: Nguyen Hoang Nhat Tan, Nguyen Chinh Quan, Luu Ha Minh Khoi

In [1]:
import pandas as pd
import numpy as np
import math

alpha for all cases = 0.05 -> z = 1.64

a)

PhD > all (income) -> All - PhD < 0 (Left-tail)

In [2]:
data = pd.read_csv(r"C:\Users\narum\Downloads\marketing_campaign.csv",sep = '\t')
data = data.dropna(subset= ["Income"])
mean_Income = data.Income.mean()
PhD_mean_Income = data[data["Education"] == "PhD"].Income.mean()
variance_Income = data.Income.var()
PhD_variance_Income = data[data["Education"] == "PhD"].Income.var()
z_Income = ((mean_Income - PhD_mean_Income) - 0 )/ math.sqrt((variance_Income/2216) + (PhD_variance_Income/481))
print(z_Income)
print(mean_Income, PhD_mean_Income, len(data[data["Education"] == "PhD"]), len(data), variance_Income, PhD_variance_Income)

-3.604814209032998
52247.25135379061 56145.31392931393 481 2216 633683788.5756189 424894944.3574931


z_income < z_alpha (-3.6 < -1.64), therefore there is evidence to say that the average income of customers with PhD is higher than 
the average of all customers.

b)

Single > Married -> Single - Married > 0 (Right-tail)

In [3]:
data = pd.read_csv(r"C:\Users\narum\Downloads\marketing_campaign.csv",sep = '\t')
data = data.dropna(subset= ["MntWines"])
single_wine = data[data["Marital_Status"] == "Single"].MntWines.mean()
married_wine = data[data["Marital_Status"] == "Married"].MntWines.mean()
single_wine_variance = data[data["Marital_Status"] == "Single"].MntWines.var()
married_wine_variance = data[data["Marital_Status"] == "Married"].MntWines.var()
z_wine = ((single_wine - married_wine) - 0)/ math.sqrt(single_wine_variance/480 + married_wine_variance/864)
print(z_wine)
print(single_wine, married_wine, len(data[data["Marital_Status"] == "Single"]), len(data[data["Marital_Status"] == "Married"]), single_wine_variance, married_wine_variance)

-0.584863735563412
288.33125 299.4803240740741 480 864 110963.1572677453 114231.83738761212


z_wine < z_alpha (-0.58 < 1.64), therefore there is no evidence to conclude that the single customers spending more on wine products 
than the married customers.

c)


masters_PhD_sober < else_sober -> masters_PhD_sober - else_sober < 0 (Left-tail)

In [4]:
data = pd.read_csv(r"C:\Users\narum\Downloads\marketing_campaign.csv",sep = '\t')
data = data.dropna(subset=  ["MntWines"])
# masters_PhD_sober = data[((data["Education"] == "PhD") | (data["Education"] == "Master")) & (data.MntWines < 50)].MntWines.mean()
# else_sober = data[((data["Education"] != "PhD") & (data["Education"] != "Master")) & (data.MntWines < 50)].MntWines.mean()
# masters_PhD_sober_var = data[((data["Education"] == "PhD") | (data["Education"] == "Master")) & (data.MntWines < 50)].MntWines.var()
# else_sober_var = data[((data["Education"] != "PhD") & (data["Education"] != "Master")) & (data.MntWines < 50)].MntWines.var()
# z_sober = ((masters_PhD_sober - else_sober) - 0) / math.sqrt(masters_PhD_sober_var/223 + else_sober_var/546)
# print(z_sober)
n1_sober = len(data[((data["Education"] == "PhD") | (data["Education"] == "Master"))].MntWines)
n2_sober = len(data[((data["Education"] != "PhD") & (data["Education"] != "Master"))].MntWines)
p1 = 223 / 856
p2 = 546 / 1384
pc = (223+546) / (856+1384)
print(223*p1, 223*(1-p1), 546*p2, 546*(1-p2))
z_sober = (p1-p2)/math.sqrt(pc*(1-pc)*((1/856)+(1/1384)))
print(z_sober)
print(n1_sober, n2_sober)
# print(masters_PhD_sober, else_sober, 
#       len(data[((data["Education"] == "PhD") | (data["Education"] == "Master")) & (data.MntWines < 50)].MntWines), 
#       len(data[((data["Education"] != "PhD") & (data["Education"] != "Master")) & (data.MntWines < 50)].MntWines),
#       masters_PhD_sober_var, else_sober_var)

58.0946261682243 164.90537383177568 215.40173410404623 330.5982658959538
-6.490032847663205
856 1384


z_sober < z_alpha (-6.49 < 1.64), therefore there is evidence to conclude that the proportion of customers with Master or PhD degrees who are sober is less than the proportion of other customers (without Master or PhD degrees) who are sober.

d)

PhD > all (MntFruits) -> All - PhD < 0 (Left-tail)

In [5]:
data = pd.read_csv(r"C:\Users\narum\Downloads\marketing_campaign.csv",sep = '\t')
data = data.dropna(subset= ["MntFruits"])
mean_Fruits = data.MntFruits.mean()
PhD_mean_Fruits = data[data["Education"] == "PhD"].MntFruits.mean()
variance_Fruits = data.MntFruits.var()
PhD_variance_Fruits = data[data["Education"] == "PhD"].MntFruits.var()
z_Fruits = ((mean_Fruits - PhD_mean_Fruits) - 0 )/ math.sqrt((variance_Fruits/2216) + (PhD_variance_Fruits/481))
print(z_Fruits)
print(mean_Fruits, PhD_mean_Fruits, len(data[data["Education"] == "PhD"]), len(data), variance_Fruits, PhD_variance_Fruits)

3.6435894719502766
26.302232142857143 20.049382716049383 486 2240 1581.9260334253172 1073.2140511645666


z_Fruits > z_alpha (3.64 > -1.64), therefore there is no evidence to say that the average amount spent on fruits products in the last 2 years of customers with PhD is higher than the average amount spent on fruits products in the last 2 years of all customers.